### PR TITLE
Improve tests to run on infrastructure

### PR DESCRIPTION
**Pull request by willallves**

### CODE REVIEW

File: infra/docker-compose.override.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#####
```
Comment (rodrigolc) [inline infra/docker-compose.override.yml:1]: OK

File: infra/spire-agent/Dockerfile [hunk 2]
```diff
@@ -1,20 +1,41 @@
 FROM ubuntu:focal
 
-RUN apt-get update \
-  && apt-get install -y --no-install-recommends ca-certificates git curl tar gzip vim unzip wget
+RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
```
Comment (rodrigolc) [inline infra/spire-agent/Dockerfile:3]: OK

File: infra/spire-server/Dockerfile [hunk 3]
```diff
@@ -1,5 +1,7 @@
 FROM ubuntu:focal
```
Comment (rodrigolc) [inline infra/spire-server/Dockerfile:1]: OK

File: infra/spire-server/create-entries.sh [hunk 4]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env bash
 set -x
```
Comment (rodrigolc) [inline infra/spire-server/create-entries.sh:2]: OK

File: infra/spire-server/init-server.sh [hunk 5]
```diff
@@ -1,2 +1,2 @@
 #!/usr/bin/env bash
```
Comment (rodrigolc) [inline infra/spire-server/init-server.sh:1]: OK

File: infra/tests/Dockerfile [hunk 6]
```diff
@@ -1,14 +1,10 @@
-FROM spire-agent
-RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
+FROM infra_workload
```
Comment (rodrigolc) [inline infra/tests/Dockerfile:1]: OK

File: infra/tests/connect_agent.sh [hunk 7]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```
Comment (rodrigolc) [inline infra/tests/connect_agent.sh:1]: OK

File: infra/tests/get-entries.sh [hunk 8]
```diff
@@ -1,2 +1,2 @@
 #!/usr/bin/env bash
-./bin/spire-server entry show
+spire-server entry show &
```
Comment (rodrigolc) [inline infra/tests/get-entries.sh:2]: OK

File: infra/tests/run-behave-tests.sh [hunk 9]
```diff
@@ -1,2 +1,2 @@
 #!/usr/bin/env bash
-behave
+behave --tags=-@wip
```
Comment (rodrigolc) [inline infra/tests/run-behave-tests.sh:2]: OK

File: infra/tests/test_install_getsvid.c [hunk 10]
```diff
@@ -0,0 +1,94 @@
+/* To compile
+ *
```
Comment (rodrigolc) [inline infra/tests/test_install_getsvid.c:2]: OK

File: infra/workload/Dockerfile [hunk 11]
```diff
@@ -8,19 +8,6 @@ ENV DEBIAN_FRONTEND noninteractive
```
Comment (rodrigolc) [inline infra/workload/Dockerfile:8]: OK

File: internal/CMakeLists.txt [hunk 12]
```diff
@@ -16,3 +15,61 @@ ${PROJECT_SOURCE_DIR}/../utils/src/util.c
 )
 
 add_library(${TARGET_NAME} SHARED ${LIB_INTERNAL})
+# Install lib:
```
Comment (rodrigolc) [inline internal/CMakeLists.txt:18]: OK

File: spiffeid/CMakeLists.txt [hunk 13]
```diff
@@ -14,3 +14,21 @@ ${PROJECT_SOURCE_DIR}/../utils/src/util.c
 )
 
 add_library(${TARGET_NAME} SHARED ${LIB_SPIFFEID})
```
Comment (rodrigolc) [inline spiffeid/CMakeLists.txt:16]: OK

File: svid/CMakeLists.txt [hunk 14]
```diff
@@ -12,3 +12,29 @@ ${PROJECT_SOURCE_DIR}/jwtsvid/src/svid.c
 ${PROJECT_SOURCE_DIR}/../utils/src/util.c)
 
 add_library(${TARGET_NAME} SHARED ${LIB_SVID})
+
```
Comment (rodrigolc) [inline svid/CMakeLists.txt:15]: OK

File: workload/CMakeLists.txt [hunk 15]
```diff
@@ -59,20 +59,64 @@ set(LIB_SOURCE
   src/jwtsource.c
 )
 
+# Install Headers:
```
Comment (rodrigolc) [inline workload/CMakeLists.txt:62]: Add jwtsource.h

File: infra/Makefile [hunk 16]
```diff
@@ -15,7 +15,7 @@ create-entries:
 	docker-compose exec spire-server ./create-entries.sh
 
 generate-token:
-	docker-compose exec spire-server spire-server token generate -spiffeID spiffe://example.org/$(SERVICE)
+	docker-compose exec spire-server spire-server token generate -spiffeID spiffe://example.org/host
```
Comment (rodrigolc) [inline infra/Makefile:18]: OK

File: workload/CMakeLists.txt [hunk 17]
```diff
@@ -59,20 +59,64 @@ set(LIB_SOURCE
...
+set(HEADERS_SOURCE
+  src/x509source.h
```
Comment (rodrigolc) [inline workload/CMakeLists.txt:64]:
```suggestion
  src/x509source.h
  src/jwtsource.h
```

**Review by rodrigolc — CHANGES_REQUESTED** [review]

**Review by rodrigolc — APPROVED** [review]

**Review by rodrigolc — CHANGES_REQUESTED** [review]
changes in workload/CMakeLists.txt

**Review by rodrigolc — APPROVED** [review]

**Review by otaviolcs3 — APPROVED** [review]

**Pull request merged**